x/stakeibc/keeper/callbacks.go

```diff
@@ -0,0 +1,155 @@
+	zone, found := k.GetHostZone(ctx, query.GetChainId())
+	if !found {
+		return fmt.Errorf("no registered zone for chain id: %s", query.GetChainId())
+	}
+	balancesStore := []byte(query.Request[1:])
+	accAddr, err := banktypes.AddressFromBalancesStore(balancesStore)
+	if err != nil {
+		return err
+	}
+
+	coin := sdk.Coin{}
+	err = k.cdc.Unmarshal(args, &coin)
+	if err != nil {
+		k.Logger(ctx).Error("unable to unmarshal balance info for zone", "zone", zone.ChainId, "err", err)
+		return err
+	}
+
+	if coin.IsNil() {
+		denom := ""
+
+		for i := 0; i < len(query.Request)-len(accAddr); i++ {
+			if bytes.Equal(query.Request[i:i+len(accAddr)], accAddr) {
+				denom = string(query.Request[i+len(accAddr):])
+				break
+			}
+
+		}
+		// if balance is nil, the response sent back is nil, so we don't receive the denom. Override that now.
+		coin = sdk.NewCoin(denom, sdk.ZeroInt())
+	}
+
+	// Set withdrawal balance as attribute on HostZone's withdrawal ICA account
+	wa := zone.WithdrawalAccount
+	wa.WithdrawalBalance = coin.Amount.Int64()
+	zone.WithdrawalAccount = wa
+	k.SetHostZone(ctx, zone)
+	k.Logger(ctx).Info(fmt.Sprintf("Just set WithdrawalBalance to: %d", wa.WithdrawalBalance))
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute("hostZone", zone.ChainId),
+			sdk.NewAttribute("totalWithdrawalBalance", coin.Amount.String()),
+		),
+	})
+
+	// Sweep the withdrawal account balance, to the feeAccount and the delegationAccount
+	//    - TODO(TEST-123) In the ICA bank send ack, create a deposit record for the swept amount that's in delegationAccount
+	k.Logger(ctx).Info(fmt.Sprintf("ICA Bank Sending %d%s from withdrawalAddr to delegationAddr.", coin.Amount.Int64(), coin.Denom))
+
+	withdrawalAccount := zone.GetWithdrawalAccount()
+	delegationAccount := zone.GetDelegationAccount()
+	// TODO(TEST-112) set the stride revenue address in a config file
+	REV_ACCT := "cosmos1wdplq6qjh2xruc7qqagma9ya665q6qhcwju3ng"
+
+	params := k.GetParams(ctx)
+	strideCommission := sdk.NewDec(int64(params.GetStrideCommission())).Quo(sdk.NewDec(100)) // convert to decimal
+	withdrawalBalance := sdk.NewDec(coin.Amount.Int64())
+	// TODO(TEST-112) don't perform unsafe uint64 to int64 conversion
+	strideClaim := strideCommission.Mul(withdrawalBalance)
+	strideClaimFloored := strideClaim.TruncateInt()
+
+	reinvestAmount := (sdk.NewDec(1).Sub(strideCommission)).Mul(withdrawalBalance)
+	reinvestAmountCeil := reinvestAmount.Ceil().TruncateInt()
+
+	// TODO(TEST-112) safety check, balances should add to original amount
+	if (strideClaimFloored.Int64() + reinvestAmountCeil.Int64()) != coin.Amount.Int64() {
+		ctx.Logger().Error(fmt.Sprintf("Error with withdraw logic: %d, Fee portion: %d, reinvestPortion %d", coin.Amount.Int64(), strideClaimFloored.Int64(), reinvestAmountCeil.Int64()))
+		panic("Failed to subdivide rewards to feeAccount and delegationAccount")
```

Comment:
Will this crash the chain? Don't think so bc SubmitQueryResponse is a standard tx but want to be sure

Reply:
don't think so; added todo to revisit in safety review